### PR TITLE
WT-4969 remove lsm_merge option from auto-generated config definitions

### PR DESCRIPTION
**Pull request by jeremytay90**

### CODE REVIEW

**Comment by wtbuild** [comment]
autobuild: Can one of the admins verify this patch?

**Comment by lukech** [comment]
test this please

**Comment by tetsuo-cpp** [comment]
@jeremytay90 PR testing failed because the `dist/s_all` script is complaining that the `__wt_config_upgrade` function always returns `0` so it could have `void` return type instead. Imo, this function should just go in the exceptions list.

File: dist/s_void [hunk 1]
```diff
@@ -124,6 +124,7 @@ func_ok()
 	    -e '/int wiredtiger_extension_init$/d' \
 	    -e '/int wiredtiger_extension_terminate$/d' \
 	    -e '/int wiredtiger_pack_close$/d' \
+		-e '/int __wt_config_upgrade$/d' \
```
Comment (tetsuo-cpp) [inline dist/s_void]: Nit: Indentation isn't consistent here.

File: src/config/config_upgrade.c [hunk 2]
```diff
@@ -15,17 +15,8 @@
...
-    config = buf->data;
-
-    /*
-     * wiredtiger_open:
-     *	lsm_merge=boolean -> lsm_manager=(merge=boolean)
-     */
-    if (__wt_config_getones(session, config, "lsm_merge", &v) != WT_NOTFOUND)
-        WT_RET(__wt_buf_catfmt(session, buf, ",lsm_manager=(merge=%s)", v.val ? "true" : "false"));
-
+    // Currently not used and included in the func_ok list of s_void
```
Comment (tetsuo-cpp) [inline src/config/config_upgrade.c]: In WiredTiger, we use `/* blah */` instead of `// blah`.

File: src/config/config_upgrade.c [hunk 3]
```diff
@@ -15,17 +15,8 @@
 int
 __wt_config_upgrade(WT_SESSION_IMPL *session, WT_ITEM *buf)
```
Comment (tetsuo-cpp) [inline src/config/config_upgrade.c]: We had a chat with @agorrod and his suggestion is to remove this function entirely. It's unlikely we'll need to symlink API options to each other in the future.

Comment (jeremytay90) [reply src/config/config_upgrade.c]: Removed this and references to this

File: src/config/config_upgrade.c (hunk 2 above)
Comment (jeremytay90) [reply src/config/config_upgrade.c]: Fixed and noted for future reference

File: dist/s_void (hunk 1 above)
Comment (jeremytay90) [reply dist/s_void]: resolved

**Pull request merged**